src/branch_view.js

```diff
@@ -23,12 +23,12 @@ function checkPreviousBanner() {
  * @param {boolean} hasApp
  */
 function renderHtmlBlob(parent, html, hasApp) {
-	journeys_utils.branch._publishEvent('willShowJourney', { 'banner_id': journeys_utils.branchViewId });
+	journeys_utils.branch._publishEvent('willShowJourney', journeys_utils.journeyLinkData);
 
 	var ctaText = hasApp ? 'OPEN' : 'GET';
 
 	journeys_utils.setPositionAndHeight(html);
-
+	journeys_utils.setJourneyLinkData(html);
```

Comment:
If you move line 26 under this, it looks like this will work.  I would love unit tests for this :)